hadoop-project/pom.xml

```diff
@@ -1387,6 +1387,20 @@
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-recipes</artifactId>
        <version>${curator.version}</version>
+       <exclusions>
+         <exclusion>
+           <groupId>org.apache.zookeeper</groupId>
+           <artifactId>zookeeper</artifactId>
```

Comment:
Note that this enables seems to be the way to enable soft-compatibility of curator 4 and ZooKeeper 3.4.